### PR TITLE
container-diff 0.7.0

### PR DESCRIPTION
**Pull request by commitay**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by ilovezfs** [comment]
@sgerrand any idea what the problem with the test is?
```
==> /usr/local/Cellar/container-diff/0.7.0/bin/container-diff analyze gcr.io/google-appengine/golang:2018-01-04_15_24 2>&1
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
Test::Unit::AssertionFailedError: <0> expected but was
<1>.
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5 
When you exit this shell, you will return to the menu.
bash-3.2$ /usr/local/Cellar/container-diff/0.7.0/bin/container-diff analyze gcr.io/google-appengine/golang:2018-01-04_15_24 
Retrieving image gcr.io/google-appengine/golang:2018-01-04_15_24 from source Cloud Registry
ERRO[0005] Error processing image: chmod /tmp/gcr.iogoogle-appenginegolang2018-01-04_15_24237399538/usr/local: operation not permitted 
bash-3.2$ echo $?
1
bash-3.2$ 
``

**Comment by sgerrand** [comment]
>
> any idea what the problem with the test is?
>
> ==> /usr/local/Cellar/container-diff/0.7.0/bin/container-diff analyze gcr.io/google-appengine/golang:2018-01-04_15_24 2>&1
> /usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
> Test::Unit::AssertionFailedError: <0> expected but was
> <1>.
> 1. raise
> 2. ignore
> 3. backtrace
> 4. irb
> 5. shell
> Choose an action: 5
> When you exit this shell, you will return to the menu.
> bash-3.2$ /usr/local/Cellar/container-diff/0.7.0/bin/container-diff analyze gcr.io/google-appengine/golang:2018-01-04_15_24
> Retrieving image gcr.io/google-appengine/golang:2018-01-04_15_24 from source Cloud Registry
> ERRO[0005] Error processing image: chmod /tmp/gcr.iogoogle-appenginegolang2018-01-04_15_24237399538/usr/local: operation not permitted
> bash-3.2$ echo $?
> 1
> bash-3.2$
>
> Not off the top of my head. I'll take a further look into this later on
tonight.

**Comment by ilovezfs** [comment]
@sgerrand thanks!

**Comment by sgerrand** [comment]
Hmm, there've been some changes in v0.7.0. Unless the image itself has changed, they shouldn't result in the error below though.

```
$ container-diff analyze gcr.io/google-appengine/golang:2018-01-04_15_24 -v info
Retrieving image gcr.io/google-appengine/golang:2018-01-04_15_24 from source Cloud Registry
INFO[0002] retrieving layer sha256:a21f4015913a12cddd687f0c0d9d908d66097da0a4085f85370c1a875d1051a2 from cache 
INFO[0005] retrieving layer sha256:729a994614f78db9504102db96281dfb2f16e9e3bb0bcf3ec229734be82ca0ee from cache 
INFO[0005] retrieving layer sha256:fe02bd96346b5291126889541d4c1ddafaf9cc0f3ac37144eeba96dbf8aa315d from cache 
ERRO[0006] Error processing image: unexpected EOF
```

**Comment by sgerrand** [comment]
Changing the Docker image reference in the test to the following should work.
```diff
- gcr.io/google-appengine/golang:2018-01-04_15_24
+ gcr.io/google-appengine/base:20180212-rc00
```

**Comment by sgerrand** [comment]
```
$ container-diff analyze gcr.io/google-appengine/base:20180212-rc00 2>&1 | egrep -e '-gcc-4.9-base\s+4.9.2-10'
-gcc-4.9-base                  4.9.2-10                        218K
```
🍻

**Comment by commitay** [comment]
@BrewTestBot test this please

File: Formula/container-diff.rb [hunk 1]
```diff
@@ -25,8 +25,8 @@ def install
   end
 
   test do
-    image = "gcr.io/google-appengine/golang:2018-01-04_15_24"
-    output = shell_output("#{bin}/container-diff analyze #{image} 2>&1")
-    assert_match /-gcc-4.9-base\s+4.9.2-10/, output
+    image = "daemon://gcr.io/google-appengine/golang:2018-01-04_15_24"
+    output = shell_output("#{bin}/container-diff analyze #{image} 2>&1", 1)
+    assert_match "Error loading image from docker engine", output
```
Comment (sgerrand) [inline Formula/container-diff.rb]: 😞 A reliable failure case is better than an unreliable assertion of success.

**Pull request closed without merging**